### PR TITLE
Pz/redemptions

### PR DESCRIPTION
**Pull request by p-zielinski**
## 20231116 - Redemptions

**Added schemas**

- Redemption
- RedemptionRollback
- RedemptionsGetResponseBody
- RedemptionsListResponseBody
- RedemptionsGetVoucherRedemptionResponseBody
- RedemptionsRollbackRequestBody
- RedemptionsRollbackResponseBody
- RedemptionRewardResult
- Sku
- Product
- RedemptionsRollbackStackableResponseBody

- GET `v1/redemptions`
  - Response body schema was replaced with new one: `RedemptionsListResponseBody` (old one: `7_res_list_redemptions`)
- GET `v1/redemptions/{redemptionId}`
  - Response body schema was replaced with new one: `RedemptionsGetResponseBody` (old one: `7_res_get_redemption`)
- GET `v1/vouchers/{code}/redemption`
  - Response body schema was replaced with new one: `RedemptionsGetVoucherRedemptionResponseBody` (old one: `7_res_get_vouchers_redemptions`)
- POST `v1/redemptions/{redemptionId}/rollback`
  - Request body schema was replaced with new one: `RedemptionsRollbackRequestBody` (old one: `7_req_rollback_redemption`)
  - Response body schema was replaced with new one: `RedemptionsRollbackResponseBody` (old one: `7_res_rollback_redemption`)
- POST `/v1/redemptions/{parentRedemptionId}/rollbacks`
  - Response body schema was replaced with new one: `RedemptionsRollbackStackableResponseBody` (old one: `19_res_redemptions_parentRedemptionId_rollbacks`)

### CODE REVIEW

File: reference/OpenAPI.json [hunk 1]
```diff
@@ -20650,34 +20853,23 @@
           }
         }
       },
-      "7_req_rollback_redemption": {
+      "RedemptionsRollbackRequestBody": {
```
Comment (jkaliszuk) [inline reference/OpenAPI.json:20856]: Tu widzę, że jest jeszcze `options` na podstawie StackedRedemptionProcessParamsOptions

File: reference/OpenAPI.json [hunk 2]
```diff
@@ -20686,9 +20878,13 @@
         },
         "description": "Request body schema for **POST** `/redemptions/{redemptionId}/rollback`."
```
Comment (jkaliszuk) [inline reference/OpenAPI.json:20879]: To chyba tez jest dla `redemptions` i `redemption`

File: reference/OpenAPI.json [hunk 3]
```diff
@@ -20686,9 +20878,13 @@
...
       },
-      "7_res_rollback_redemption": {
+      "RedemptionsRollbackResponseBody": {
         "title": "Rollback Redemption Response Body",
-        "$ref": "#/components/schemas/7_obj_rollback_redemption_object_extended",
+        "allOf": [
```
Comment (jkaliszuk) [inline reference/OpenAPI.json:20883]: Czy tu musi byc `allOf` jezeli jest tylko jeden schemat?

File: reference/OpenAPI.json [hunk 4]
```diff
@@ -29407,6 +29643,132 @@
           }
         }
       },
+      "Sku": {
```
Comment (jkaliszuk) [inline reference/OpenAPI.json:29646]: Po co nowe obiekty dla SKU i PRODUCT. To chyba to samo co SkuResponse i ProductResponse zrobione przezemnie. Tez widze ze w Product np. deleted chyba nie jest zwracane.

**Review by jkaliszuk — CHANGES_REQUESTED** [review]

File: reference/OpenAPI.json (hunk 1 above)
Comment (p-zielinski) [reply reference/OpenAPI.json:20856]: Oznaczony komentarzem jako `deprecated` więc nie dodaje

File: reference/OpenAPI.json (hunk 2 above)
Comment (p-zielinski) [reply reference/OpenAPI.json:20879]: nie

**Review by jkaliszuk — APPROVED** [review]

**Pull request merged**